internal/controllers/clusterextension_controller.go

```diff
@@ -682,32 +687,43 @@ func clusterExtensionRequestsForCatalog(c client.Reader, logger logr.Logger) crh
 	}
 }
 
-// getInstalledVersion fetches the installed version of a particular bundle from the cluster. To do so, we read the release label
-// which are added during install.
-func (r *ClusterExtensionReconciler) getInstalledVersion(ctx context.Context, clusterExtension ocv1alpha1.ClusterExtension) (*bsemver.Version, error) {
-	cl, err := r.ActionClientGetter.ActionClientFor(ctx, &clusterExtension)
+func (r *ClusterExtensionReconciler) installedBundle(ctx context.Context, allBundles []*catalogmetadata.Bundle, ext *ocv1alpha1.ClusterExtension) (*catalogmetadata.Bundle, error) {
+	cl, err := r.ActionClientGetter.ActionClientFor(ctx, ext)
 	if err != nil {
 		return nil, err
 	}
 
-	release, err := cl.Get(clusterExtension.GetName())
+	release, err := cl.Get(ext.GetName())
 	if err != nil && !errors.Is(err, driver.ErrReleaseNotFound) {
 		return nil, err
 	}
 	if release == nil {
 		return nil, nil
 	}
 
-	existingVersion, ok := release.Labels[labels.BundleVersionKey]
-	if !ok {
-		return nil, fmt.Errorf("release %q: missing bundle version", release.Name)
+	// Bundle must match installed version exactly
+	vr, err := mmsemver.NewConstraint(release.Labels[labels.BundleVersionKey])
+	if err != nil {
+		return nil, err
 	}
 
-	existingVersionSemver, err := bsemver.New(existingVersion)
-	if err != nil {
-		return nil, fmt.Errorf("could not determine bundle version for the chart %q: %w", release.Name, err)
+	// TODO (dfranz) Are the PackageName + BundleName + BundleVersion enough to
+	// be 100% sure we have the correct bundle?
```

Comment:
Please see `TODO` here.

Reply:
I think so, the catalog is clusters scoped - I don't think having same bundle in multiple catalogs makes sense.

Reply:
@m1kola could you please confirm if that's right?

Reply:
The code has been modified as before, @dtfranz can we resolve this then?